### PR TITLE
[MM-15105] Add Go code samples to API reference for plugin endpoints

### PR DESCRIPTION
**Pull request by z4cco**
Solves ticket : [MM-15105](https://mattermost.atlassian.net/browse/MM-15105)

Fixes https://github.com/mattermost/mattermost-server/issues/10609

### CODE REVIEW

File: v4/source/plugins.yaml [hunk 1]
```diff
@@ -102,6 +134,17 @@
...
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/platform/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
+
+            pluginID := "4xp9fdt77pncbef59f4k1qe83o"
+
+            ok, resp = Client.RemovePlugin(pluginID)
```
Comment (hanzei) [inline v4/source/plugins.yaml]: Plugin ID's are actually more meaningful the other id's in Mattermost. Let's use a real world example here:
```suggestion
            ok, resp = Client.RemovePlugin("com.mattermost.demo-plugin")
```

File: v4/source/plugins.yaml [hunk 2]
```diff
@@ -164,6 +218,17 @@
...
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/platform/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
+
+            pluginID := "4xp9fdt77pncbef59f4k1qe83o"
+
+            ok, resp = Client.DisablePlugin(pluginID)
```
Comment (hanzei) [inline v4/source/plugins.yaml]:
```suggestion
            ok, resp = Client.DisablePlugin("com.mattermost.demo-plugin")
```

File: v4/source/plugins.yaml [hunk 3]
```diff
@@ -133,6 +176,17 @@
...
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/platform/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
+
+            pluginID := "4xp9fdt77pncbef59f4k1qe83o"
+
+            ok, resp = Client.EnablePlugin(pluginID)
```
Comment (hanzei) [inline v4/source/plugins.yaml]:
```suggestion
            ok, resp = Client.EnablePlugin("com.mattermost.demo-plugin")
```

File: v4/source/plugins.yaml [hunk 4]
```diff
@@ -192,3 +257,12 @@
           $ref: '#/responses/Forbidden'
         '501':
           $ref: '#/responses/NotImplemented'
+      x-code-samples:
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/platform/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
```
Comment (hanzei) [inline v4/source/plugins.yaml]: This requires no authentication.

File: v4/source/plugins.yaml (hunk 1 above)
Comment (z4cco) [reply v4/source/plugins.yaml]: done

File: v4/source/plugins.yaml (hunk 2 above)
Comment (z4cco) [reply v4/source/plugins.yaml]: done

File: v4/source/plugins.yaml (hunk 3 above)
Comment (z4cco) [reply v4/source/plugins.yaml]: done

File: v4/source/plugins.yaml (hunk 4 above)
Comment (z4cco) [reply v4/source/plugins.yaml]: done

**Pull request merged**